src/odin_devices/dac63004.py

```diff
@@ -0,0 +1,392 @@
+            #write the flipped value
+            self.bus.write_word_data(self.address, reg, int("0b" + value, 2))
+        except IOError as err:
+            logging.error("--------------------------------------------------")
+            logging.error("Error: " + str(err))
+            logging.error("Write16 failed to write value " + str(bin(value)) + " to register " + str(hex(reg)))
+            logging.error("--------------------------------------------------")
+            return -1
+        
+    def readU16flipped(self, reg):
+        """Read an unsigned 16-bit value from the I2C device, replacing the first 8 bits with the last 8 bits and vice versa
+        
+        reg (int): the address of the register to read
+        """
+        try:
+            result = self.bus.read_word_data(self.address, reg)
+            #convert the read value to binary
+            result = str(bin(result)).replace("0b", "")
+            #make sure the read value is 16 bits long
+            while (len(result) < 16):
+                result = "0" + result
+            #flip the first and last 8 bits around
+            result = result[8:] + result[:8]
+            #convert the result back to denary
+            return int("0b" + result, 2)
+        except IOError as err:
+            logging.error("--------------------------------------------------")
+            logging.error("Error: " + str(err))
+            logging.error("Read16 failed to read value from register " + str(hex(reg)))
+            logging.error("--------------------------------------------------")
+            return -1
+
+    def set_dac_as_current(self, index, voltagePowerDown="11"):
+        """ Set the dac at the provided index into current output mode, powering down the voltage mode in the method specified
+        
+        voltagePowerDown (string):
+            "01": Power-down VOUT-X with 10 KΩ to AGND
+            "10": Power-down VOUT-X with 100 KΩ to AGND
+            "11": Power-down VOUT-X with Hi-Z to AGND
+        """
+        #There are only 4 dacs, so check there is a valid index
+        if (index >= 0 and index <= 3):
+            # Build the mask and value to write based on the index of the dac we want to enable. 
+            # To set a dac to current, we want to write 110 to the appropriate section on the COMMON_CONFIG register.
+            # The 11 sets the voltage output for the dac to Hi-Z power down mode
+            # The 0 sets the current output for the dac to powered up mode
+            # See page 61 in the data sheet for more info on the COMMON_CONFIG register
+            mask = "0000" + ("000" * (3-index)) + "111" + ("000" * index)
+            write = "0000" + ("000" * (3-index)) + voltagePowerDown + "0" + ("000" * index)
+            self.read_modify_write("COMMON_CONFIG", int(mask, 2), int(write, 2))
+        else:
+            print("Not a valid index.")
+            return
+        
+    def set_dac_as_voltage(self, index):
```

Comment:
Consider name change for clarity; this is setting a mode, rather than a numeric voltage/current.